phoenix-core/src/main/java/org/apache/phoenix/index/GlobalIndexChecker.java

```diff
@@ -586,17 +595,28 @@ private boolean verifyRowAndRepairIfNecessary(List<Cell> cellList) throws IOExce
                 long ts = cellList.get(0).getTimestamp();
                 cellList.clear();
 
+                long repairTime;
                 try {
                     repairIndexRows(rowKey, ts, cellList);
+                    repairTime = EnvironmentEdgeManager.currentTimeMillis() - repairStart;
                     metricsSource.incrementIndexRepairs(indexName);
                     metricsSource.updateUnverifiedIndexRowAge(indexName,
                         EnvironmentEdgeManager.currentTimeMillis() - ts);
                     metricsSource.updateIndexRepairTime(indexName,
                         EnvironmentEdgeManager.currentTimeMillis() - repairStart);
+                    if (shouldLog()) {
+                        LOG.info(String.format("Index row repair on region %s took %d ms.",
+                                env.getRegionInfo().getRegionNameAsString(), repairTime));
+                    }
```

Comment:
This is good one, however regardless of `shouldLog()`, we might want to consider logging at TRACE level always:
```
                    if (shouldLog()) {
                        LOG.info("Index row repair on region {} took {} ms.",
                                env.getRegionInfo().getRegionNameAsString(), repairTime);
                    }
                    if (LOG.isTraceEnabled()) {
                        LOG.trace("Index row repair on region {} took {} ms.",
                                env.getRegionInfo().getRegionNameAsString(), repairTime);
                    }
```

Reply:
nit: `String.format()` is not required, we can use slf4j log placeholder `{}`
```
                        LOG.info("Index row repair on region {} took {} ms.",
                                env.getRegionInfo().getRegionNameAsString(), repairTime);
```

Reply:
Agree on the placeholder, but I don't think addig the unconditional trace log is a good idea.
We should not use one set of conditions for INFO, and another for TRACE, and should not double log, as both conflicts with the usual/expected logging semantics.

Reply:
I was just thinking if a user enables TRACE logging, this is a good log candidate for all read repairs (it might be overwhelming too but that's why unconditional TRACE might be more suitable) and whereas % based logs at INFO level won't make it overwhelming which is the purpose of this PR anyways. Thoughts?

Reply:
On one hand, logging every event with trace would make sense from an operational standpoint, as it can theoretically be enabled without restarting the RS (I THINK that LOG4j will pick logging config file changes during runtime)

On the other hand I feel that having a statistic logging at INFO level, but also logging everything at TRACE level would go against the 'do not surprise/confuse the user' principle. AFAIK we have other statistic logging events, which cover similar cases, but we don't override the percentage setting like this.

IMO **REPAIR_LOGGING_PERCENT_ATTRIB sets the probability for logging the event at INFO, but we're logging every event at TRACE anyway** is not intuitive behaviour, nor is logging the same event twice at different logging levels.

Unfiltered trace logging in production systems in unsustainable, and if someone wants to do targeted log index rebuilds, he can just use the percentage setting (which is easier to get right than logging config, even if it does require an RS restart to take effect)

Reply:
> nor is logging the same event twice at different logging levels.

Yeah got your point, sure I think it's fine, we don't need TRACE level logging, anyways we have some useful metrics available too e.g `unverifiedIndexRowAge`, `indexRepairTime` etc